### PR TITLE
services/horizon: Export ledger stats in /metrics, add claimable_balance to change stats

### LINKED ISSUE #3133: Add ledger stats to /metrics
Expose all `stats_` metrics collected during ledger ingestion to `/metrics`. This will help correlate slower ingestion with activity type in the ledger.

### PR DESCRIPTION
**Pull request by bartekn**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [x] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [x] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://www.stellar.org/developers/reference/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [ ] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [x] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

Export ledger stats (number of ops and changes) in `/metrics`. Add claimable balances to changes stats.

Close #3133.

### Why

To be able to correlate slow ledger with network activitity.

### CODE REVIEW

File: ingest/io/stats_change_processor.go [hunk 1]
```diff
@@ -82,6 +95,10 @@ func (stats *StatsChangeProcessorResults) Map() map[string]interface{} {
 		"stats_accounts_updated": stats.AccountsUpdated,
 		"stats_accounts_removed": stats.AccountsRemoved,
 
+		"stats_claimable_balances_created": stats.ClaimableBalancesCreated,
+		"stats_claimable_balances_updated": stats.ClaimableBalancesUpdated,
+		"stats_claimable_balances_removed": stats.ClaimableBalancesRemoved,
```
Comment (Shaptic) [inline ingest/io/stats_change_processor.go:100]: Do we want to extend the CB integration tests to check these values?

Comment (bartekn) [reply ingest/io/stats_change_processor.go:100]: I'm not sure we have any tests for this so far. If so, I'd vote for adding this in another PR.

File: services/horizon/internal/ingest/fsm.go [hunk 2]
```diff
@@ -471,9 +473,25 @@ func (r resumeState) run(s *system) (transition, error) {
...
+			With(prometheus.Labels{"type": stat}).Add(float64(value.(int64)))
+	}
+
+	ledgerTransactionStatsMap := ledgerTransactionStats.Map()
+	for stat, value := range ledgerTransactionStatsMap {
+		stat = strings.Replace(stat, "stats_", "ledger_", 1)
+		s.Metrics().LedgerStatsCounter.
+			With(prometheus.Labels{"type": stat}).Add(float64(value.(int64)))
+	}
+
```
Comment (2opremio) [inline services/horizon/internal/ingest/fsm.go:491]: would it be worth to factor out the metrics-updating code? (the method is already pretty large)

Comment (bartekn) [reply services/horizon/internal/ingest/fsm.go]: @2opremio do you have something like https://github.com/stellar/go/pull/3148/commits/d6f557b481ae31e1170feb94d058045945d30151 in mind?

Comment (2opremio) [reply services/horizon/internal/ingest/fsm.go]: That's better, yeah.

**Comment by bartekn** [comment]
@2opremio @Shaptic can I have a final 👍 ?

**Review by 2opremio — APPROVED** [review]

**Pull request merged**